src/scripting/lua_root.cc

```diff
@@ -802,6 +807,12 @@ void LuaDescriptions::do_modify_tribe(lua_State* L,
 		tribe_descr.mutable_wares().insert(di);
 		tribe_descr.mutable_wares_order()[column].push_back(di);
 
+		Widelands::WareDescr& wdescr = *descrs.get_mutable_ware_descr(di);
+		if (!lua_isnil(L, 7)) {
+			wdescr.set_default_target_quantity(unit_name, luaL_checkuint32(L, 7));
+		}
+		wdescr.set_preciousness(unit_name, luaL_checkuint32(L, 8));
```

Comment:
what if  L,8 is nil? will luaL_checkuint32 handle this? Perhaps would be better tocheck for !isnil here as well

Reply:
All `lua[L]_check...` functions perform type checking. If the 8th argument is nil or not a number it will report a coroutine error `"bad argument #8 to <FUNCTION>: number expected, got <TYPE>"`.

The reason why I allow nil for `#7` but not for `#8` is that preciousness should be mandatory while target quantity is optional

Reply:
Well,  in this case it would be helpful for the addon designer to provide a proper error message as for e.g. the worker. I always found it hard to debug these bad argument errors in my lua code.